### PR TITLE
Bump doctrine/common dependency

### PR DESCRIPTION
**Pull request by jmikola**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,2 @@
 phpunit.xml
-vendor
-composer.lock
```
Comment (stof) [inline .gitignore]: why commiting the lock file in the library ?

Comment (jmikola) [inline .gitignore]: I talked with @naderman about this at FOSDEM. It's ignored for projects that depend on ODM, so it doesn't affect our users, but it gives developers consistency for reproducing test results (also the "build status" icon).

**Comment by BernardoSilva** [comment]
Commiting the lock file will force dependencies to be locked for that specific version.
If you're locking the dependencies to a specific revision it can cause issues if we depend directly on `doctrine/mongodb`.

**Comment by naderman** [comment]
Composer does not consider the content of a dependency's lock file, only the lock file in the root project itself is ever used.

**Pull request merged**